lib/puppet/parser/functions/hiera_array.rb

```diff
@@ -27,7 +27,7 @@ module Puppet::Parser::Functions
   If any matched value is a hash, puppet will raise a type mismatch error.
 
   More thorough examples of `hiera` are available at:  
-  <http://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
+  <https://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
```

Comment:
Probably best to use this link now - https://docs.puppetlabs.com/hiera/3.0/puppet.html#hiera-lookup-functions

Example will need checked though.